### ISSUE TITLE
Wizard: Break features and pages out

### ISSUE THREAD

**Issue by crutkas** [Microsoft/WindowsTemplateStudio#607]
We've hit that time now where we need to break pages and features out.  Too many options for a single page IMO.

**Comment by mrlacey** [comment]
split over two steps in the wizard (getting to 3 in total)?
Pages, then features seems sensible to me.

**Comment by crutkas** [comment]
Yes.  Navigation & Framework -> Pages -> Features.

Once xamarin comes into play, we'll need maybe to rethink the first step as step 1 then would be Platforms

**Comment by ralarcon** [comment]
Do you think this can wait until 1.2 is delivered?

**Comment by crutkas** [comment]
I marked it for 1.3 when i created it.  Just wanted to create the work item.  Also call out the fact we'll need to think about the xamarin adjustment for platform :)

**Comment by mrlacey** [comment]
A good candidate to bump to 1.4 as only added 1 feature and no pages in 1.3 - @crutkas?

**Comment by crutkas** [comment]
agreed.  lets push to 1.4 but IMO this will be needed for xam + wizard update

**Comment by mvegaca** [comment]
I'm thinking about start working on it tomorrow morning.

**Comment by mrlacey** [comment]
Should the splitting of ProjectType and Framework (& the XF option) be added at the same time? They seem very closely related and if major changes will need to be made for this any such changes should also consider what will be needed for that. I can't find an issue for that work but I know it's been discussed. Making sure this is fully understood will be necessary before any work can begin.

There's also a potential overlap with #875 and #827 that should probably also be considered at the same time even if not implemented together.

**Comment by mvegaca** [comment]
I think that project type and framework should keep together in step 1.
I think that re-categorize features are only related to the concept the templates, categories are composed using a static field on templates json.
Redesign adding new pages and features can be a different task, my idea was to work on this keeping the same code and UX

**Comment by mrlacey** [comment]
/cc @crutkas Re: "I think that project type and framework should keep together in step 1." I know there has been lots of discussion on splitting this up. Partly for the XF work and partly to better support project types that aren't supported by some frameworks (which I know was being discussed for Prism). If changes are being made to the wizard they should consider the other planned changes too so anything done now doesn't make things harder for the known future changes if this can be avoided.

If the redesign and UI changes are large and are also going to happen for the next release then doing them separately will likely lead to unnecessary and wasted effort spent making changes and keeping the current UI. Whether this is an issue depends on how much time is available to be spent on this and how big the planned changes are.

**Comment by crutkas** [comment]
I think #827 without a doubt should be done here and the work here should be done in concert with the redesign in mind but not exclusively.  I think redesign will happen in in the Xamarin timeframe as well as we need to allow for a much more robust system with the all permutations this will provide.

The current adjustments all have pages and features broken out so this adjustment is minimal.

TLDR: all changes should think of the rippling effects throughout.

**Comment by mvegaca** [comment]
Implemented on #1135.
The selection now is on two different step pages but the code behind is shared like current solution. Added some base ViewModel refactor and moved infer naming code to a different ViewModel to reduce TemplatesViewModel

Tested on extension (VS Experimental instance)

**Comment by sibille** [comment]
Verified in dev-nightly: 
Wizard: 0.14.17282.65001
Templates: 0.14.17282.1
